bndtools.core/src/bndtools/launch/BndDependencySourceContainer.java

```diff
@@ -65,13 +67,17 @@ public ISourceContainerType getType() {
         List<ISourceContainer> result = new LinkedList<ISourceContainer>();
 
         ILaunchConfiguration config = getLaunchConfiguration();
+        Set<String> projectsAdded = new HashSet<>();
         try {
             Project project = LaunchUtils.getBndProject(config);
             if (project != null) {
                 Collection<Container> runbundles = project.getRunbundles();
                 for (Container runbundle : runbundles) {
                     if (runbundle.getType() == TYPE.PROJECT) {
                         String targetProjName = runbundle.getProject().getName();
+                        if (projectsAdded.contains(targetProjName))
+                            continue;
+                        projectsAdded.add(targetProjName);
```

Comment:
This would be better as:
```java
if (projectsAdded.add(targetProjName)) {
   IProject targetProj = ...
}
```

That is use the boolean return value from `add` to decide whether to add the JavaProjectSourceContainer instead of a separate `contains` check.